internal/host/refresh_status_preprocessor.go

```diff
@@ -194,8 +193,41 @@ func newValidations(v *validator) []validation {
 			condition: v.sufficientOrUnknownInstallationDiskSpeed,
 			formatter: v.printSufficientOrUnknownInstallationDiskSpeed,
 		},
+		{
+			id:        HasSufficientNetworkLatencyRequirementForRole,
+			condition: v.hasSufficientNetworkLatencyRequirementForRole,
+			formatter: v.printSufficientNetworkLatencyRequirementForRole,
+		}, {
+			id:        HasSufficientPacketLossRequirementForRole,
+			condition: v.hasSufficientPacketLossRequirementForRole,
+			formatter: v.printSufficientPacketLossRequirementForRole,
+		},
 	}
-	return ret
+	return filterHostValidations(disabledHostValidations, v.log, baseValidations)
+}
+
+func filterHostValidations(disabledHostValidations []string, log logrus.FieldLogger, validations []validation) []validation {
+	f := funk.Filter(validations, func(v validation) bool {
+		for _, id := range disabledHostValidations {
```

Comment:
WDYT about pre-processing `disabledHostValidations` into a set (`map[string]bool`) so during filtering the code wouldn't have to iterate over `disabledHostValidations` for each element in `validations`?

Reply:
Makes things easier definitely, but the burden is now in the definition in the environment variable, where it now looks like a map/dict:
```
DISABLED_HOST_VALIDATIONS=validation-1:true,validaton-2:true
```
It just looks odd this way. One could think that the boolean value determines whether the validation is disabled or enabled (false meaning not disabled/enabled)....

Maybe in the future we can migrate out from plain environment variables and use a similar approach to what you did for the resource requirements for OCP. Maybe there's room to migrate some of the variables in the `Options` struct into a more structured way that doesn't rely in envconfig.

Reply:
What I had in mind is pre-processing the slice of strings into a `map[string]bool` at the beginning of the function - to have one iteration plus little memory overhead instead of `len(validations)` iterations over `disabledHostValidations`.

The other option might be having a dedicated envconfig decoder that transforms env var string into a map (like here: https://github.com/openshift/assisted-service/blob/69594371fa01cd67db43093542ee5f6fbd449134/internal/operators/cnv/config.go#L14).

Reply:
Thanks! I like the `Decode` function, I'll check that path :smile:

Reply:
So I went the `Decode` path and it looks much better now. I ended up moving the variable into the `host.Config` structure :smile: